src/GameConfiguration/index.ts

```diff
@@ -79,12 +106,74 @@ export default class GameConfiguration implements GameConfiguration {
   }
 
   update(visibleGameState: VisibleGameInformation): this {
+    const { grid, tiles } = this.revealed
+    const { passable, unknownObstacles, adjacencies, distances, cities, crowns } = this.hidden
+
+    function tileAt(rowIndex: number, colIndex: number): Tile | undefined {
+      const row = grid[rowIndex]
+      return row && row[colIndex]
+    }
+
+    function maybeAddAdjacency(tile: Tile, otherTile: Tile | undefined): void {
+      if (otherTile && passable.has(otherTile)) {
+        adjacencies.get(tile)!.add(otherTile)
+        adjacencies.get(otherTile)!.add(tile)
+        distances.get(tile)!.set(otherTile, 1)
```

Comment:
We're always setting distances between two tiles together - let's make a function for this and update the areas where we set distances